src/modules/components/transaction/transactionDataListItem/transactionDataListItemStructure/transactionDataListItemStructure.tsx

```diff
@@ -0,0 +1,109 @@
+import {
+    AvatarIcon,
+    DataList,
+    Heading,
+    IconType,
+    NumberFormat,
+    Spinner,
+    formatterUtils,
+    type AvatarIconVariant,
+} from '../../../../../core';
+import { QueryType, createBlockExplorerLink } from '../../../../utils/blockExplorerUtils/blockExplorerUtils';
+import { formatDate } from '../../../../utils/timestampUtils';
+import {
+    TransactionType,
+    TxStatusCode,
+    txHeadingStringList,
+    txIconTypeList,
+    txVariantList,
+    type ITransactionDataListItemProps,
+} from './transactionDataListItemStructure.api';
+
+export const TransactionDataListItemStructure: React.FC<ITransactionDataListItemProps> = (props) => {
+    const {
+        chainId,
+        tokenAddress,
+        tokenSymbol,
+        tokenValue,
+        usdEstimate,
+        txType,
+        txStatus = TxStatusCode.PENDING,
+        unixTimestamp,
+        txHash,
+        ...otherProps
+    } = props;
+
+    const getEffectiveStatus = () => {
+        const type = txType;
+        if (txStatus === TxStatusCode.FAILED) {
+            return {
+                icon: IconType.CLOSE,
+                variant: 'critical' as AvatarIconVariant,
+                heading: 'Failed transaction',
+            };
+        }
+
+        return {
+            icon: type ? txIconTypeList[type] : IconType.HELP,
+            variant: type ? txVariantList[type] : ('neutral' as AvatarIconVariant),
+            heading: type ? txHeadingStringList[type] : 'Unknown',
+        };
+    };
+
+    const { icon, variant, heading } = getEffectiveStatus();
+
+    const getTxIcon = () => {
+        if (txStatus !== TxStatusCode.PENDING) {
+            return <AvatarIcon className="shrink-0" variant={variant} icon={icon} responsiveSize={{ md: 'md' }} />;
+        }
+        return (
+            <div className="flex size-6 shrink-0 items-center justify-center md:size-8">
+                <Spinner className="transition" variant="neutral" responsiveSize={{ md: 'lg' }} />
+            </div>
+        );
+    };
+
+    return (
+        <DataList.Item
+            className="min-w-fit !py-0 px-4 md:px-6"
```

Comment:
The className specified here will be overwritten by the otherProps being spread afterwards. You can use the classNames function to combine that and the className coming from the props. Also, by doing so, you shouldn't neeed to set the padding as important.